### PR TITLE
f64: Refactor collapsible_if

### PR DESCRIPTION
**Pull request by wcampbell0x2a**

### CODE REVIEW

**Comment by shepmaster** [comment]
Can you explain why this refactoring is beneficial?

**Comment by wcampbell0x2a** [comment]
> Can you explain why this refactoring is beneficial?

This make it more apparent that the other `else if` clauses are also part of the alternative part of the first conditional `if !cfg!(any(target_os = "solaris", target_os = "illumos")) {` instead of being part of another nested `if`.

File: library/std/src/f64.rs [hunk 1]
```diff
@@ -920,22 +920,20 @@ impl f64 {
...
-            } else if self > 0.0 {
-                self // log(Inf) = Inf
+        } else if self.is_finite() {
+            if self > 0.0 {
+                log_fn(self)
+            } else if self == 0.0 {
+                Self::NEG_INFINITY // log(0) = -Inf
             } else {
-                Self::NAN // log(-Inf) = NaN
+                Self::NAN // log(-n) = NaN
```
Comment (camelid) [inline library/std/src/f64.rs:929]: Is this definitely correct? Just want to double-check.

Comment (jyn514) [reply library/std/src/f64.rs:929]: @camelid the diff is misleading, this is the same case as `self.is_finite()` from before. I recommend viewing with whitespace changes disabled and you'll see there's no logic changed.

Comment (camelid) [reply library/std/src/f64.rs:929]: You're right, that *is* a very misleading diff.

**Pull request merged**